name-service/js/test/e2e/index.test.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  Connection,
+  Keypair,
+  LAMPORTS_PER_SOL,
+  PublicKey,
+  sendAndConfirmTransaction,
+  Transaction,
+} from '@solana/web3.js';
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import {
+  createNameRegistry,
+  deleteNameRegistry,
+  getHashedName,
+  getNameAccountKey,
+  NameRegistryState,
+  reallocNameAccount,
+  transferNameOwnership,
+  updateNameRegistryData,
+} from '../../src';
+
+chai.use(chaiAsPromised);
+const url = 'http://localhost:8899';
+
+describe('Name Service Program', () => {
+  let connection: Connection;
+  let payer: Keypair;
+  let owner: Keypair;
+  let name: string;
+  let nameKey: PublicKey;
+  before(async () => {
+    connection = new Connection(url, 'confirmed');
+    payer = Keypair.generate();
+    const airdropSignature = await connection.requestAirdrop(
+      payer.publicKey,
+      4 * LAMPORTS_PER_SOL
+    );
+    await connection.confirmTransaction(airdropSignature, 'confirmed');
+    name = '.sol';
+    nameKey = await getNameKey(name);
+  });
+  it('Create Name Registery', async () => {
+    const space = 10;
+    owner = Keypair.generate();
+    const lamports = await connection.getMinimumBalanceForRentExemption(
+      space + NameRegistryState.HEADER_LEN
+    );
+    const inst = await createNameRegistry(
+      connection,
+      name,
+      space,
+      payer.publicKey,
+      owner.publicKey,
+      lamports
+    );
+    const tx = new Transaction().add(inst);
+    await sendAndConfirmTransaction(connection, tx, [payer]);
+    const nameAccount = await NameRegistryState.retrieve(connection, nameKey);
+    nameAccount.owner.equals(owner.publicKey);
+    expect(nameAccount.data?.length).to.eql(10);
+  });
+  it('Update Name Registery', async () => {
```

Comment:
Could you refactor all these tests so they don't depend on each other? Right now, for example, this test assumes that `name` has an existing registry, so instead could you create a new registry for each test as needed?  You could have it in the `before` clause